forc-util/src/fs_locking.rs

```diff
@@ -136,11 +139,43 @@ impl PidFileLocking {
         fs.flush()?;
         Ok(())
     }
+
+    /// Cleans up all stale lock files in the .lsp-locks directory
+    /// Returns a vector of paths that were cleaned up
+    pub fn cleanup_stale_files() -> io::Result<Vec<PathBuf>> {
+        let lock_dir = user_forc_directory().join(".lsp-locks");
+        let entries = read_dir(&lock_dir)?;
+        let mut cleaned_paths = Vec::new();
+
+        for entry in entries {
+            let entry = entry?;
+            let path = entry.path();
+            if let Some(ext) = path.extension().and_then(|ext| ext.to_str()) {
+                if ext == "lock" {
+                    if let Ok(mut file) = File::open(&path) {
+                        let mut contents = String::new();
+                        if file.read_to_string(&mut contents).is_ok() {
+                            if let Ok(pid) = contents.trim().parse::<usize>() {
+                                if !Self::is_pid_active(pid) {
+                                    remove_file(&path)?;
+                                    cleaned_paths.push(path);
+                                }
+                            } else {
```

Comment:
I don't think the `else` should do clean up as there's a race condition - the other process could be in the middle of writing it's PID into the lockfile itself 😅